util-core/src/main/scala/com/twitter/util/Promise.scala

```
@@ -574,38 +574,70 @@ class Promise[A] extends Future[A] with Promise.Responder[A] with Updatable[Try[
-          s.handler.applyOrElse(intr, Promise.AlwaysUnit)
-        else {
-          val current = Local.save()
-          if (current ne s.saved)
-            Local.restore(s.saved)
-          try s.handler.applyOrElse(intr, Promise.AlwaysUnit)
-          finally Local.restore(current)
+  @tailrec final def raise(intr: Throwable): Unit = {
+    val funcName = "raise"
+    CoverageChecker.initialize(funcName, 15)
+    state match {
+      case waitq: WaitQueue[A] =>
+        if (!cas(waitq, new Interrupted(waitq, intr))) {
+          CoverageChecker.reached(funcName, 0)
+          raise(intr)
+        } else {
+          CoverageChecker.reached(funcName, 1)
         }
-      }
-
-    case s: Transforming[A] =>
-      if (!cas(s, new Interrupted(s.waitq, intr))) raise(intr)
-      else {
-        s.other.raise(intr)
-      }
-
-    case s: Interrupted[A] =>
-      if (!cas(s, new Interrupted(s.waitq, intr)))
-        raise(intr)
-
-    case _: Try[A] /* Done */ => () // nothing to do, as its already satisfied.
-
-    case p: Promise[A] /* Linked */ => p.raise(intr)
+  
+      case s: Interruptible[A] =>
+        if (!cas(s, new Interrupted(s.waitq, intr))) {
+          CoverageChecker.reached(funcName, 2)
+          raise(intr)
+        } else {
+          CoverageChecker.reached(funcName, 3)
+          if (!UseLocalInInterruptible) {
+            CoverageChecker.reached(funcName, 4)
+            s.handler.applyOrElse(intr, Promise.AlwaysUnit)
+          } else {
+            CoverageChecker.reached(funcName, 5)
+            val current = Local.save()
+            if (current ne s.saved) {
+              CoverageChecker.reached(funcName, 6)
+              Local.restore(s.saved)
+            } else {
+              CoverageChecker.reached(funcName, 7)
+            }
+            try {
+              s.handler.applyOrElse(intr, Promise.AlwaysUnit)
+            } catch {
+              case _: Throwable => CoverageChecker.reached(funcName, 8) 
```

This will swallow the exception